utils/mvn/utils.go

```diff
@@ -47,13 +48,33 @@ func RunMvn(configPath, deployableArtifactsFile string, buildConf *utils.BuildCo
 	return mvnRunConfig.runCmd()
 }
 
-func validateMavenInstallation() error {
+func getMavenHome() (string, error) {
 	log.Debug("Checking prerequisites.")
 	mavenHome := os.Getenv(MavenHome)
 	if mavenHome == "" {
-		return errorutils.CheckError(errors.New(MavenHome + " environment variable is not set"))
+		// The M2_HOME environment variable is not defined.
+		// Since Maven installation can be located in different locations,
+		// depending on the installation type and the OS (for example: For Mac with brew install: /usr/local/Cellar/maven/{version}/libexec or Ubuntu with debian: /usr/share/maven),
+		// we need to grab the location using the mvn --version command
+		log.Debug(MavenHome, " is not defined. Retrieving Maven home using mvn --version.")
+		cmd := exec.Command("mvn", "--version")
+		var stdout bytes.Buffer
+		cmd.Stdout = &stdout
+		err := errorutils.CheckError(cmd.Run())
+		if err != nil {
+			return "", err
+		}
+		output := strings.Split(strings.TrimSpace(stdout.String()), "\n")
+		// Finding the relevant "Maven home" line in command response.
+		for _, line := range output {
+			if strings.HasPrefix(line, "Maven home:") {
+				mavenHome = strings.Split(line, " ")[2]
+				break
+			}
+		}
 	}
-	return nil
```

Comment:
I think we should check here if mavenHome is still empty, and in this case, return an error sayong that we failed to fetch the maven home from the 'mvn --version' output. Worth adding the actual command output to the error message.